core/src/Track.cc

```diff
@@ -1415,20 +1413,22 @@ bool Track::checkConsistency() const {
   }
 
   if (trackPointsWithMeasurement.size() != trackPointsWithMeasurement_.size()) {
-    errorOut << "Track::checkConsistency(): trackPointsWithMeasurement_ has incorrect size" << std::endl;
-    retVal = false;
+    failures << "Track::checkConsistency(): trackPointsWithMeasurement_ has incorrect size" << std::endl;
+    consistent = false;
   }
 
   for (unsigned int i = 0; i < trackPointsWithMeasurement.size(); ++i) {
     if (trackPointsWithMeasurement[i] != trackPointsWithMeasurement_[i]) {
-      errorOut << "Track::checkConsistency(): trackPointsWithMeasurement_ is not correct" << std::endl;
-      errorOut << "has         id " << i << ", address " << trackPointsWithMeasurement_[i] << std::endl;
-      errorOut << "should have id " << i << ", address " << trackPointsWithMeasurement[i] << std::endl;
-      retVal = false;
+      failures << "Track::checkConsistency(): trackPointsWithMeasurement_ is not correct" << std::endl;
+      failures << "has         id " << i << ", address " << trackPointsWithMeasurement_[i] << std::endl;
+      failures << "should have id " << i << ", address " << trackPointsWithMeasurement[i] << std::endl;
+      consistent = false;
     }
   }
 
-  return retVal;
+  if (not consistent) {
```

Comment:
As just discussed, we will keep consistent variable because there is one case where the stringstream is filled, but the consistent variable not changed to false.